### ISSUE TITLE
Deep-copying graphs interferes with $.

### ISSUE THREAD

**Issue by merkys** [graphviz-perl/Graph#26]
MWE:
```perl
open( my $inp, $0 );
while( <$inp> ) {
    print "BEFORE $.\n";
    my $g = Graph->new;
    $g->add_path( 'A', 'B', 'C', 'D' );
    my $h = $g->deep_copy;
    print "AFTER  $.\n";
}
close $inp;
```
Expected output:
```
BEFORE 1
AFTER  1
BEFORE 2
AFTER  2
...
```
Actual output:
```
BEFORE 1
AFTER  0
BEFORE 2
AFTER  2
```
Thus some code in `deep_copy` must interfere with `$.`. Strangely, its value is restored on the second iteration of the cycle. To me it seems that the issue appeared in version 0.9705.

**Comment by mohawk2** [comment]
Thanks! Released fix as 0.9724.

**Comment by merkys** [comment]
Great, thanks a lot for prompt fix!
